vcr/test/oidc4vci_integration_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package test
+
+import (
+	"context"
+	"github.com/golang/mock/gomock"
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/test"
+	httptest "github.com/nuts-foundation/nuts-node/test/http"
+	"github.com/nuts-foundation/nuts-node/vcr"
+	"github.com/nuts-foundation/nuts-node/vcr/api/oidc4vci_v0"
+	"github.com/nuts-foundation/nuts-node/vcr/holder"
+	"github.com/nuts-foundation/nuts-node/vcr/issuer"
+	vcrTypes "github.com/nuts-foundation/nuts-node/vcr/types"
+	"github.com/nuts-foundation/nuts-node/vdr/didservice"
+	"github.com/nuts-foundation/nuts-node/vdr/types"
+	"github.com/stretchr/testify/require"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+var issuerDID = did.MustParseDID("did:nuts:GvkzxsezHvEc8nGhgz6Xo3jbqkHwswLmWw3CYtCm7hAW")
+var receiverDID = did.MustParseDID("did:nuts:B8PUHs2AUHbFF1xLLK4eZjgErEcMXHxs68FteY7NDtCY")
+var credentialID = ssi.MustParseURI(issuerDID.String() + "#1")
+var credential = vc.VerifiableCredential{
+	Context: []ssi.URI{
+		didservice.JWS2020ContextV1URI(),
+		didservice.NutsDIDContextV1URI(),
+	},
+	ID: &credentialID,
+	Type: []ssi.URI{
+		ssi.MustParseURI("VerifiableCredential"),
+		ssi.MustParseURI("NutsAuthorizationCredential"),
+	},
+	Issuer:       issuerDID.URI(),
+	IssuanceDate: time.Now().Truncate(time.Second),
+	CredentialSubject: []interface{}{map[string]interface{}{
+		"ID": receiverDID.String(),
+	}},
+}
+
+// TestOIDC4VCIHappyFlow tests issuing a VC using OIDC4VCI.
+// Steps:
+//   - Create Echo server, register OIDC4VCI API
+//   - Issue a VC using the OIDC4VCI Credential Issuer, check that it is received by the wallet
+//   - Check that the VC is stored in the wallet
+func TestOIDC4VCIHappyFlow(t *testing.T) {
```

Comment:
This is a weird test because it uses an echo server but not with our own middleware. What is the benefit of using an http server over testing the wrapper with mocks?

Reply:
I rewrote the test to use an actual started Nuts node